internal/etcd/etcd.go

```diff
@@ -149,6 +151,78 @@ func (c *ClientEtcdAPI) Close() error {
 	return nil
 }
 
+// UsedSpacePercentage returns the used space as a percentage of the storage quota
+func (c *ClientEtcdAPI) UsedSpacePercentage(ctx context.Context, member Member) (int, error) {
+	if len(member.ClientURLs) < 1 {
+		return 0, fmt.Errorf("unable to get used space, etcd member has no client urls")
+	}
+	url := member.ClientURLs[0] + "/metrics"
+	resp, err := http.Get(url)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get metrics from etcd: %w", err)
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read metrics returned from etcd: %w", err)
+	}
+	defer resp.Body.Close()
+
+	const (
+		storageQuotaKey = "etcd_server_quota_backend_bytes"
+		storageSizeKey  = "etcd_mvcc_db_total_size_in_bytes"
+	)
+
+	strs := strings.Split(string(b), "\n")
+	var metricsWeCareAbout []string
+	for _, str := range strs {
+
+		if strings.HasPrefix(str, storageQuotaKey) ||
+			strings.HasPrefix(str, storageSizeKey) {
+			metricsWeCareAbout = append(metricsWeCareAbout, str)
+		}
+		// We are only looking for 2 metrics
+		if len(metricsWeCareAbout) >= 2 {
+			continue
+		}
+	}
```

Comment:
This also feels like it could be made more generic somehow and extracted to its own function?

Reply:
As part of separating into `GetMetrics`, I've cleaned this up